### DISCUSSION TITLE
Using Pulsar Plus with the EM112 Energy Meter

### DISCUSSION THREAD

**Discussion by GrantKeymer** [lbbrhzn/ocpp#281] [General]
Happy New Year to all,

Recently I've been trying to use my Wallbox Pulsar Plus with the Carlo Gavazzi EM112 Energy Meter.  This meter connects via RS485 to the Pulsar Plus and when detected, is supposed to allow some additional functions...

- Power Boost
- ECO-Smart

These are enabled by options in the Upgrades menu of the Wallbox app.

These so-called 'Upgrades' are quite primitive compared to what is possible with HA, but in theory they should be a simple way of achieving load-balancing between charger and household (Power Boost), or maximising the use of Solar Power (ECO-Smart).

The problem is, neither of these options seem to work, at least for me.

Just wondering if anyone else has tried to use the EM112, or a similar energy meter with their Pulsar Plus?

Cheers,
Grant.

**Comment by lbbrhzn** [comment]
I have a Pulsar Plus with EM340. I do have the Power Boost enabled, but not ECO-Smart.  I knew Wallbox was planning to introduce ECO-Smart, but I had not noticed it was already there! According to the [information](https://support.wallbox.com/en/knowledge-base/how-to-use-the-eco-smart-feature/) on the Wallbox site, the minimum charging power in Eco Mode is 460 W per phase. My solar panels deliver 3 kW peak, much less in the winter, so I don't really have the opportunity to try that now. Also, I noticed that ECO-Smart is ignored for manual and scheduled charges. 

For Power Boost you need set the max current for your household. I've got it set to 25A, and this seems to work fine. It should only start limiting when I have multiple appliances running, which is seldom the case. I'll try setting it to a lower value to see if it limits the charging current accordingly.

**Comment by GrantKeymer** [comment]
@lbbrhzn Thanks very much for your reply.  After looking at the info you linked on the Wallbox site, I see that it has recently been updated.  Here is the original version...

![Original Eco Smart Document](https://user-images.githubusercontent.com/35156198/147862029-1f04dc56-1e57-4531-8663-4eefdd86a1f9.png)

**_And here is the updated version..._**

![Modified Eco Smart document](https://user-images.githubusercontent.com/35156198/147862036-f144a191-4506-4205-acdc-18330ee12605.png)

Now I can see that Wallbox have discontinued support of Eco-Smart for Copper SB but more importantly, they have drastically shortened the list of Energy Meters that are supported.  Now it seems EM530 is the required model.

My local Wallbox distributor mentioned there was some updated info about Eco-Smart but then went on holiday without telling me what it was.  Thanks to your help, now I know.  It's also interesting to know about the minimum 460W per phase option which wasn't mentioned before.  Somebody was asking about that in another thread, so I'll post the details there.

Well, at least something good has come out of this update.  EM530 will not fit in my switchboard, so I will return the EM112 which never worked anyway and concentrate on developing an automation in HA.

What is the hardware platform you are using to run HA?

I really need to get something that works instead of wasting time with Windows.

**Reply by lbbrhzn** [reply]
Currently I'm using a Qotom Q510G6 with a Celeron 3865U.  I've installed Proxmox VE on it, with Home Assistant OS running on a virtual machine. 

Previously I used a dedicated Raspberry PI 4 for HA, which worked fine as well, but it was quite a bit slower. The Home Assistant Blue / Yellow seem to be good alternatives as well.

**Reply by GrantKeymer** [reply]
Thanks for the info.  Now I have a few options to look at.

Probably leaning towards Raspberry Pi 4B with 4GB RAM at this point as I'm only expecting to run a single automation for the Pulsar Plus.  I guess that's what everybody thinks at the start and then they end up automating more and more of their house.

**Reply by andre-richter** [reply]
@GrantKeymer I am a bit confused about the `EM530` only thing. When you go to the NA website, it is the only EM that is offered in their store.

On the German website, for example, they still offer the EM340: https://wallbox.com/de_de/dreiphasiger-energiezahler

No sight of the EM530. Something country specific?

**Reply by lbbrhzn** [reply]
On the dutch site they also don't mention the EM350, only EM112 or EM330, EM340 version PF.B.
See https://support.wallbox.com/nl/knowledge-base/hoe-je-de-eco-smart-functie-moet-gebruiken/

**Reply by GrantKeymer** [reply]
@andre-richter - Yes, you are right.  After further checking, that document I found is in the NA part of the Wallbox website.  Strange that it appeared when I clicked on the link that @lbbrhzn posted yesterday.  If I click on that same link now, it goes to an updated version of the original document, with mention of Eco mode at 460W per phase.  Interesting to know that the German and Dutch versions of that document still reference the same list of energy meters.

Sorry for the confusion.

**Comment by lbbrhzn** [comment]
I can confirm that the Pulsas Plus + EM340 does wait for an energy surplus when Eco-Smart is activated. @andre-richter

**Comment by lbbrhzn** [comment]
In addition I can confirm Powerboost is pausing the charging when the demand + current usage is larger than the maximum set in the app. Also, the charging current is varied dynamically, depending on the current usage.

**Reply by GrantKeymer** [reply]
Great work, many thanks @lbbrhzn.

**Reply by lbbrhzn** [reply]
I wonder if you might have installed your EM112 incorrectly. In my system the EM340 is measuring the net current consumption of the complete household, including charger. Is that also how you have installed it?

**Reply by GrantKeymer** [reply]
@lbbrhzn - In my case, installing the EM112 is complicated.  If I can get it to work, it will need to be in another building about 200 metres away, where the solar panels and inverter are located, rather than in the garage at our house, where the Pulsar Plus is located.  I will need to buy some WiFi to RS485 adapters, install another WiFi access point, etc, etc.  Together with electrician's charges for changing the switchboard wiring, the cost would be well over 1000 Euros.

Before committing to that, I wanted to make sure it would work, hence my temporary test setup as shown in the video below...

https://drive.google.com/file/d/1y2jXiRCWxOZfXXwKmfMWknN6rpqmAzPJ/view?usp=sharing

I can simulate import of energy from the grid, or export to the grid by connecting the EM112 input and output terminals appropriately, and changing the amount of power used by a test load.  That's why I found it very strange that the MyWallbox app would never let me enable Power Boost or Eco-Smart, even though my Pulsar Plus has been communicating with the EM112, even though it was only for about 30 seconds.

After that disappointment, it made me think about this a lot more, hence I'm considering going down the HA route, with OCPP rather than using the Pulsar Plus to make decisions about how much charge to put into my EV.  For one thing, I already have an Energy Meter in my switchboard, installed by the Solar Power company.  It provides the following information, updated every minute...

![Solar Power Status](https://user-images.githubusercontent.com/35156198/147892830-a47299bb-10cb-473b-97aa-00f257433e07.png)

If I already have that information, there must be a way to access it, either by scraping the webpage or by API.

Secondly, having access to ALL the power flow data, including house battery charging power, allows me to make smarter decisions, for example...

Reduce house battery charging power to 2kW and divert the other 2kW into the EV, because I can fill up the house battery later in the afternoon, at a rate less than 1.2kW.

If I only have access to grid power (import and export), then I don't have the opportunity to divert some of the house battery charging power.  I know it will take a lot more work than simply connecting an EM112, but I should get a much better result, which I can tailor to my needs.  As you can tell, I enjoy tinkering :)

**Reply by drc38** [reply]
@GrantKeymer Evcc has just released Ocpp compatibility https://github.com/evcc-io/evcc/issues/1139#issuecomment-1005951745 if you try it out let me know how you get on. I've not yet got solar installed but am planning to this year.

**Reply by analystcmyk** [reply]
A bit late but you might have a look at debugging the raw modbus values of meter using relyd/modbussniffer 
If the initial handshake is not correct between the meter and the wallbox, it will just disable powerboost and ecosmart. Reaching out to wallbox they considered it not "necessary" to have  access to the raw data in the UI, so you could debug, let alone sending it over MQTT.
